### ISSUE TITLE
[BUG] kubelet prober metrics on k8s 1.25+

### ISSUE THREAD

**Issue by a7i** [DataDog/datadog-agent#17523]
Upgrading to Kubernetes 1.25 or later, the following metrics are no longer reported: 
`kubernetes.readiness_probe.success.*`
`kubernetes.liveness_probe.success.*`

Kubernetes 1.25 Metrics queried via kubelet prober prometheus endpoint:
```
prober_probe_total{container="app-nginx",namespace="app",
pod="app-nginx",pod_uid="2de634c5-963f-4ffb-9b15-700f3f9b8bf3",
probe_type="Readiness",result="successful"
} 794
```

Kubernetes 1.24 Metric queried via kubelet prober prometheus endpoint:
```
prober_probe_total{container="app-nginx",namespace="app",
pod="app-nginx-799b74869b-mmhr7",pod_uid="4550420b-6a0c-4d59-a486-f5b3a69ebdf8",
probe_type="Readiness",result="successful"
} 81023
```

This breaking change causes the metrics to not get processed by [datadog core](https://github.com/DataDog/integrations-core/blob/22899496fd0908344f1db0920d55554645b85578/kubelet/datadog_checks/kubelet/probes.py#L94) which relies on actual pod name.

**Agent Environment**

`v7.40.0`

**Describe what happened:**
Upstream Issue: https://github.com/kubernetes/kubernetes/issues/115766
The [root cause](https://github.com/kubernetes/kubernetes/issues/115766#issuecomment-1446747588) is the [PR](https://github.com/kubernetes/kubernetes/pull/104484) that landed in 1.25

**Describe what you expected:**
The metrics continue to show up, even on k8s 1.25+

**Steps to reproduce the issue:**
- run with k8s 1.25
- observe that metrics don't show up

**Additional environment details (Operating System, Cloud provider, etc):**
Kubernetes 1.25
Cloud provider: AWS
OS: Ubuntu 22.04

**Comment by dliffredo** [comment]
Hi any update?
we saw that also kubernetes.readiness_probe.failure.* are no longer reported

**Comment by a7i** [comment]
fyi, my upstream PR was merged: https://github.com/kubernetes/kubernetes/pull/104484
and backported to 1.25, 1.26, 1.27 (and included in 1.28)

**Comment by clamoriniere** [comment]
Thanks @a7i for the update
